### PR TITLE
Dispatch MM unit test workflows based on modular-magician downstream SHAs

### PR DESCRIPTION
**Pull request by shuyama1**
Following the [Cloud Build Guide](https://cloud.google.com/build/docs/configuring-builds/pass-data-between-steps#passing_data_using_workspaces) to pass upstream commit sha between build steps

fixes https://github.com/hashicorp/terraform-provider-google/issues/17924

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: .ci/magician/github/set.go
Comment (shuyama1) [inline .ci/magician/github/set.go:107]: This is only for testing the updated tgc-test workflow in this PR. Will need to revert to "main" before merging.

**Review by melinath — CHANGES_REQUESTED** [review]

File: .ci/magician/cmd/test_tpg.go
Comment (melinath) [inline .ci/magician/cmd/test_tpg.go:87]: It looks like the workflow dispatch for the tests failed here with a 422 error; possibly some issue with this change?

**Review by melinath — CHANGES_REQUESTED** [review]

File: .ci/magician/cmd/test_tpg.go
Comment (shuyama1) [reply .ci/magician/cmd/test_tpg.go:87]: Nice catch! Sorry I didn't realize that the dispatch failed and the tests were not triggered. It was running successfully before I added back the pr-number input. Let me look into it

File: .ci/magician/cmd/generate_downstream.go
Comment (shuyama1) [reply .ci/magician/cmd/generate_downstream.go:319]: Totally. I'll update the variable names.

File: .ci/magician/cmd/test_tpg.go
Comment (melinath) [reply .ci/magician/cmd/test_tpg.go:87]: yeah.... it shouldn't fail quietly in that case. I think that's a holdover from when we didn't want TGC compilation errors to be a blocker. But that aspect isn't something we need to figure out in this PR...

Comment (shuyama1) [reply .ci/magician/cmd/test_tpg.go:87]: It also quietly failed for tpg and tpgb tests, which is more concerning IMO

Comment (shuyama1) [reply .ci/magician/cmd/test_tpg.go:87]: We'll need a separate PR that contains the dispatch workflow configuration changes to test this PR. Therefore, I created the branch directly against the main https://github.com/GoogleCloudPlatform/magic-modules/pull/10675 a couple of weeks ago and mentioned in https://github.com/GoogleCloudPlatform/magic-modules/pull/10671/files#r1598095588. However, I got myself confused and closed it last night somehow. I'll reopen and update that PR and then trigger tests in this PR. Sorry for the confusion.

Comment (melinath) [reply .ci/magician/cmd/test_tpg.go:87]: The dispatch workflow should always run using the main branch - it might be necessary to make a backwards-compatible change to the workflow first, then make this PR second

Comment (shuyama1) [reply .ci/magician/cmd/test_tpg.go:87]: yeah, I can create a backwards-compatible change to the workflow first, but we'll still need to create a separate branch to test that change if I understand the testing process correctly.

Comment (shuyama1) [reply .ci/magician/cmd/test_tpg.go:87]: nvm, I think I understand what's the ask now

Comment (melinath) [reply .ci/magician/cmd/test_tpg.go:87]: I was thinking it would be necessary to test the changes in this PR but I see you've actually already worked around that with the line that makes it use this PR's code for the dispatch 👍

File: .ci/magician/cmd/test_tgc.go
Comment (melinath) [inline .ci/magician/cmd/test_tgc.go:50]:
```suggestion
	contentTPGB, err := os.ReadFile("/workspace/commitSHA-terraform-provider-google-beta.txt")
```

File: .ci/magician/cmd/test_tgc.go
Comment (melinath) [inline .ci/magician/cmd/test_tgc.go:63]: Is this accurate, or is there a case when the commit SHA is expected to be unknown?

```suggestion
	// fall back to branch if commit SHA can't be found
	if commitShaOrBranchUpstreamTPGB == "" {
```

File: .ci/magician/cmd/test_tpg.go
Comment (melinath) [inline .ci/magician/cmd/test_tpg.go:76]: would be nice to have a comment here as well.

File: .github/workflows/test-tgc.yml
Comment (melinath) [reply .github/workflows/test-tgc.yml:20]: ping on this one

File: .github/workflows/test-tgc.yml
Comment (melinath) [inline .github/workflows/test-tgc.yml:23]:
```suggestion
        description: 'The branch or sha of the tgc execute against'
```

**Review by melinath — CHANGES_REQUESTED** [review]
a couple small changes to request

**Comment by melinath** [comment]
@shuyama1 could you look at this today?

**Comment by melinath** [comment]
If not, just choose a new reviewer.

File: .ci/magician/cmd/test_tgc.go
Comment (shuyama1) [reply .ci/magician/cmd/test_tgc.go:63]: I think this is accurate. I honestly can't think of a case where the magician commit is generated but the commit SHA is unknown. It usually will be just no commit is generated and we'll need fall back to branch. 

I'll add the comment.

File: .github/workflows/test-tgc.yml
Comment (shuyama1) [reply .github/workflows/test-tgc.yml:20]: oops, I modified the tpgb branch description. Will update it!

File: mmv1/third_party/terraform/website/docs/r/apigee_flowhook.html.markdown
Comment (shuyama1) [inline mmv1/third_party/terraform/website/docs/r/apigee_flowhook.html.markdown:87]: will revert before merging

File: mmv1/third_party/tgc/tests/data/bucket.json
Comment (shuyama1) [inline mmv1/third_party/tgc/tests/data/bucket.json:3]: will revert before merging

File: mmv1/third_party/tgc/tests/data/bucket.tf
Comment (shuyama1) [inline mmv1/third_party/tgc/tests/data/bucket.tf:31]: will revert before merging

File: mmv1/third_party/tgc/tests/data/bucket.json
Comment (shuyama1) [inline mmv1/third_party/tgc/tests/data/bucket.json:3]: will revert before merging

**Comment by shuyama1** [comment]
@melinath Please take a look when you have time. I'll revert the test changes in .ci/magician/github/set.go, mmv1/third_party/terraform/website/docs/r/apigee_flowhook.html.markdown, mmv1/third_party/tgc/tests/data/bucket.json and mmv1/third_party/tgc/tests/data/bucket.tf before merging the PR. Thanks!

**Comment by melinath** [comment]
@shuyama1 It looks like the tests didn't run for the latest commit for some reason? Maybe the test branch needs to be updated?

**Comment by shuyama1** [comment]
> @shuyama1 It looks like the tests didn't run for the latest commit for some reason? Maybe the test branch needs to be updated?

I think the tests didn't run due to the files I touched are not an effective go files, so the actions are triggered but actual tests are skipped https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/9686675108/job/26729632752?pr=10671. I double checked the commit sha of tpg, tpgb and tgc in the [presubmit-generate-diffs](https://pantheon.corp.google.com/cloud-build/builds/68c77fba-a53a-434c-a279-df28d5096ded?project=graphite-docker-images&e=-13802955&invt=AbVHBA&inv=1) and they did match with the latest version of the commits in magician branches. 

I'll make a change to some go files that will actually trigger the tests later.

**Comment by shuyama1** [comment]
@melinath I kicked off a run to trigger all the tests. tgc-unit-test's failure is probably due to the change I made to the tgc test files (?). I checked the commit sha in the generate-dfifs and they do match with the three latest generated commits in the magician's repos

**Review by melinath — APPROVED** [review]
LGTM (pending revert of the changed code to run tests) - thanks!

**Comment by shuyama1** [comment]
I'll revert the testing code and merge the PR later today

**Comment by shuyama1** [comment]
@melinath I've reverted all the test-related changes. I plan to merge this PR later today unless there are any objections. Thanks for reviewing!

**Pull request merged**
